src/bundle/Controller/TabController.php

```diff
@@ -0,0 +1,66 @@
+<?php
+/**
+ * File containing the TabController class.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\HybridPlatformUiBundle\Controller;
+
+use eZ\Publish\API\Repository\ContentService;
+use eZ\Publish\Core\MVC\Symfony\Controller\Controller;
+use EzSystems\HybridPlatformUi\Http\Response\ResetResponse;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\Routing\RouterInterface;
+
+abstract class TabController extends Controller
+{
+    /**
+     * @var RouterInterface
+     */
+    protected $router;
+
+    /**
+     * @var ContentService
+     */
+    protected $contentService;
+
+    public function __construct(
+        RouterInterface $router,
+        ContentService $contentService
+    ) {
+        $this->router = $router;
+        $this->contentService = $contentService;
+    }
+
+    protected function resetLocation($contentId)
+    {
+        $mainLocationId = $this->contentService->loadContentInfo($contentId)->mainLocationId;
+
+        return new ResetResponse($this->generateContentViewUrl($contentId, $mainLocationId));
+    }
+
+    protected function reloadTab($tab, $contentId, $locationId)
+    {
+        return $this->redirectToTab($contentId, $locationId, $tab . '_tab');
+    }
+
+    private function redirectToTab($contentId, $locationId, $viewType)
```

Comment:
This feels like it duplicates `reloadTab`, and the `RedirectResponse` could be instanciated inside that method instead.